testsV2/ut_repyv2api_virtualnamespacecontextsafety.py

```diff
@@ -1,21 +1,29 @@
 """
-This unit test checks that VirtualNamespace.evaluate() forbids an unsafe context.
+Verify that VirtualNamespace.evaluate() forbids an unsafe `context`.
+
+Specifically, the `context` that a VirtualNamespace object receives 
+must be a `SafeDict`, or parseable as one. This means that the 
+context's keys must be of type `str` (which we try to violate in this 
+test), avoid particular names / prefixes / components, etc.
+
+Using an unsafe context should raise a `ContextUnsafeError` when 
+evaluatinig the VirtualNamespace.
+
+See `safe.py` for what exactly a `SafeDict` is and isn't allowed to 
+do and to contain.
 """
 
 #pragma repy 
```

Comment:
`utf.py` doesn't really care, it looks only at the `#pragma`s to figure out what to do. Overall, we probably should rename all Repy tests to `.r2py`, but that's (again) a different issue: SeattleTestbed/utf#76